README.md

```diff
@@ -2,107 +2,112 @@
-
-
 ## Usage
 
-To use this library in a Cairo contract:
-1. Include a copy of `accesscontrol_library.cairo` in your project, and import the library into the Cairo contract.
-2. Define the available roles as constants in a namespace in a separate Cairo contract, and import this namespace into the Cairo contract.
+To use this library, add the repository as a dependency in your `Scarb.toml`:
 
-For example, assuming you have a `contracts/` folder with `accesscontrol_library.cairo` and `roles.cairo`, and you want to import both into a Cairo file within the same folder:
+```
+[dependencies]
+wadray = { git = "https://github.com/lindy-labs/cairo-wadray.git" }
+```
 
+Next, define the available roles in a separate Cairo file:
 ```cairo
-%lang starknet
-
-from starkware.cairo.common.cairo_builtins import BitwiseBuiltin, HashBuiltin
-from contracts.accesscontrol_library import AccessControl
-from contracts.roles import Roles
-
-@view
-func is_manager{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*
-}(user: felt) -> (authorized: felt) {
-    let authorized: felt = AccessControl.has_role(Roles.MANAGER, user);
-    return (authorized,);
+mod user_roles {
+    const MANAGER: u128 = 1;
+    const STAFF: u128 = 2;
+    const USER: u128 = 4;
 }
+```
+then import both the component and the roles into your Cairo contract.
+
+For example, assuming you have a `src/` folder with the roles defined in a `user_roles` module in `roles.cairo`:
+```
+use starknet::ContractAddress;
 
-@external
-func authorize{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*}(
-    role: felt, user: felt
-) {
-    AccessControl.assert_admin();
-    AccessControl._grant_role(role, user);
-    return ();
+#[starknet::interface]
+trait IMockContract<TContractState> {
+    fn is_manager(self: @TContractState, user: ContractAddress) -> bool;
 }
 
-@external
-func manager_only_action{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr, bitwise_ptr: BitwiseBuiltin*}() {
-    AccessControl.assert_has_role(Roles.MANAGER);
-    // Insert logic here
-    return ();
+#[starknet::contract]
+mod mock_contract {
+    use access_control::access_control::access_control_component;
+    use src::roles::user_roles;
+    use starknet::ContractAddress;
+    use super::IMockContract;
+
+    component!(path: access_control_component, storage: access_control, event: AccessControlEvent);
+
+    #[abi(embed_v0)]
+    impl AccessControlPublic = access_control_component::AccessControl<ContractState>;
+    impl AccessControlHelpers = access_control_component::AccessControlHelpers<ContractState>;
+
+    #[storage]
+    struct Storage {
+        #[substorage(v0)]
+        access_control: access_control_component::Storage
+    }
+
+    #[event]
+    #[derive(Copy, Drop, starknet::Event, PartialEq)]
```

Comment:
Nit - the `PartialEq` is necessary only for testing right? If so, I'd remove it from here.